tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -1660,31 +1660,27 @@ Status StorageManager::load_array_metadata(
   const auto& array_metadata_to_load = array_dir.array_meta_uris();
```

Comment:
We might be able to easily convert StorageManager::load_array_metadata to throw instead of returning Status as part of this PR.

Reply:
done